src/tags/object/RichText/domManager.ts

```diff
@@ -298,52 +333,100 @@ class DomData {
     this.elements.push(dynamicBlock);
     return dynamicBlock;
   }
+
   setDisplayedText(displayedText: string) {
     this.displayedText = displayedText;
   }
-  addStaticElement(currentNode: HTMLElement, path:Path) {
+
+  addStaticElement(currentNode: HTMLElement, path: Path) {
     this.elements.push(new DDStaticElement(currentNode, this.endPos, path.toString()));
   }
-  addExtraText(text:DDExtraText) {
+
+  addExtraText(text: DDExtraText) {
     let lastIdxOfTextBlock = this.elements.length - 1;
 
     while (!(this.elements[lastIdxOfTextBlock] instanceof DDDynamicBlock) && lastIdxOfTextBlock > -1) {
       --lastIdxOfTextBlock;
     }
     this.elements.splice(lastIdxOfTextBlock + 1, 0, text);
   }
-  addTextElement(textNode: Text, path:Path) {
+
+  findProjectionOnDisplayedText(text: string) {
+    const { displayedText } = this;
+    let fromIdx = this.displayedTextPos;
+    const contentParts = [];
+
+    while (displayedText[fromIdx] === LF) {
+      fromIdx++;
+    }
+    let toIdx = fromIdx;
+
+    for (const char of text) {
+      if (displayedText[toIdx] === char || (displayedText[toIdx] === ' ' && char === LF)) {
+        contentParts.push(displayedText[toIdx]);
+        toIdx++;
+      } else {
+        contentParts.push('');
+      }
+    }
+    return {
+      fromIdx,
+      toIdx,
+      content: contentParts.map(parts => parts ? [...parts] : parts).flat(),
+    };
+  }
+
+  addTextElement(textNode: Text, path: Path) {
     const { displayedText } = this;
-    const text:string = textNode.textContent as string;
-    const length = [...(text)].length;
-    const pos = displayedText.indexOf(text, this.displayedTextPos);
+    const text: string = textNode.textContent as string;
+    let pos = displayedText.indexOf(text, this.displayedTextPos);
+    let content = [...(text)];
+    const contentLength = content.length;
+    let displayedTextLength = text.length;
+
+    if (pos === -1) {
+      // text doesn't match any parts of displayedText
+      // that means that it contains some \n or other symbols that are trimmed by browser
+
+      // calc the offsets of the part of displayedText that matches the text in terms of displayed symbols
+      const { fromIdx, toIdx, content: newContent } = this.findProjectionOnDisplayedText(text);
+
+      pos = fromIdx;
+      displayedTextLength = toIdx - fromIdx;
+      // fill content with dummies for not displayable symbols
+      content = newContent;
+    }
```

Comment:
This is an important find, awesome work! I did notice this type of issue within Paragraphs as well, and we have a similar type of logic there. I think after seeing this work, we probably should look to reuse this DomManager there as well, and remove any unnecessary logic from the Paragraphs. Not to be done here, but a future improvement and code reuse.